tensorboard/webapp/widgets/histogram/histogram_linked_time_fob_test.ts

```diff
@@ -0,0 +1,182 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+import {NO_ERRORS_SCHEMA} from '@angular/core';
+import {ComponentFixture, TestBed} from '@angular/core/testing';
+import {By} from '@angular/platform-browser';
+import {
+  Fob,
+  LinkedTimeFobControllerComponent,
+} from '../linked_time_fob/linked_time_fob_controller_component';
+import {LinkedTime} from '../linked_time_fob/linked_time_types';
+import {TemporalScale} from './histogram_component';
+import {HistogramLinkedTimeFobController} from './histogram_linked_time_fob_controller';
+
+describe('HistogramLinkedTimeFobController', () => {
+  let temporalScaleSpy: jasmine.Spy;
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      declarations: [
+        HistogramLinkedTimeFobController,
+        LinkedTimeFobControllerComponent,
+      ],
+      schemas: [NO_ERRORS_SCHEMA],
+    }).compileComponents();
+  });
+
+  function createComponent(input: {
+    steps?: number[];
+    linkedTime?: LinkedTime;
+  }): ComponentFixture<HistogramLinkedTimeFobController> {
+    const fixture = TestBed.createComponent(HistogramLinkedTimeFobController);
+    fixture.componentInstance.steps = input.steps ?? [100, 200, 300, 400];
+    fixture.componentInstance.linkedTime = input.linkedTime ?? {
+      start: {step: 200},
+      end: null,
+    };
+    temporalScaleSpy = jasmine.createSpy();
+    fixture.componentInstance.temporalScale =
+      temporalScaleSpy as unknown as TemporalScale;
+    temporalScaleSpy.and.callFake((step: number) => {
+      // Imitate a 10 to 1 scale.
+      return step * 10;
+    });
+    return fixture;
+  }
+
+  it('returns first element of steps from getLowestStep', () => {
+    let fixture = createComponent({steps: [100, 200, 300, 400]});
+    expect(fixture.componentInstance.getLowestStep()).toBe(100);
+  });
+
+  it('returns final element of steps from getHighestStep', () => {
+    let fixture = createComponent({steps: [100, 200, 300, 400]});
+    expect(fixture.componentInstance.getHighestStep()).toBe(400);
+  });
+
+  describe('getStepHigherThanAxisPosition', () => {
+    it('gets step higher when position is not on a step', () => {
+      let fixture = createComponent({steps: [100, 200, 300, 400]});
+      let stepHigher =
+        fixture.componentInstance.getStepHigherThanAxisPosition(1500);
+      expect(stepHigher).toEqual(200);
+    });
+    it('gets step on given position when that position is on a step', () => {
```

Comment:
Please add new lines between each `it('...'`
Here and most places in this test file.